File: src/main/java/EmojiBot.java
import org.json.JSONObject;
import org.telegram.telegrambots.bots.TelegramLongPollingBot;
import org.telegram.telegrambots.meta.api.methods.send.SendMessage;
import org.telegram.telegrambots.meta.api.objects.Update;
import org.telegram.telegrambots.meta.exceptions.TelegramApiException;

import javax.net.ssl.HttpsURLConnection;
import java.io.*;
import java.net.URL;
import java.util.List;
import java.util.Locale;
import java.util.Random;

public class EmojiBot extends TelegramLongPollingBot {

    @Override
    public void onUpdateReceived(Update update) {
        SendMessage message = new SendMessage();
        message.setChatId(update.getMessage().getChatId().toString());

        if (update.hasMessage() && update.getMessage().hasText()) {

            String[] array = update.getMessage().getText().split(" ");

            StringBuilder b = new StringBuilder();
            StringBuilder append = new StringBuilder();

            try {
                URL url = new URL("https://raw.githubusercontent.com/farkmarnum/emojify/main/src/data/emoji-data.json");
                HttpsURLConnection connection = (HttpsURLConnection) url.openConnection();
                connection.setRequestMethod("GET");
                BufferedReader reader = new BufferedReader(new InputStreamReader((InputStream) connection.getContent()));
                String s;
                while ((s = reader.readLine()) != null) {
                    append.append(s.trim());
                }
                connection.disconnect();
            } catch (IOException e) {
                e.printStackTrace();
            }

            for (String value : array) {
                StringBuilder modifiedTxt = setModifiedText(value, append);
                b.append(modifiedTxt);
            }

            message.setText(b.toString());
            try {
                execute(message);
            } catch (TelegramApiException ignored) {
            }

        }
    }

    private StringBuilder setModifiedText(String value, StringBuilder jsonData) {
        StringBuilder appendText = new StringBuilder();
        JSONObject obj = new JSONObject(jsonData.toString());

        try {
            JSONObject object = obj.getJSONObject(value.toLowerCase(Locale.ROOT));
            String hold = value + " " + object.names().get(getRandomPos(object.length())) + " ";
            return appendText.append(hold);
        } catch (Exception e) {
            e.printStackTrace();

        }

        return new StringBuilder(value + " ");
    }

    private int getRandomPos(int length) {
        Random rand = new Random();
        return rand.nextInt(length);
    }

    @Override
    public void onUpdatesReceived(List<Update> updates) {
        super.onUpdatesReceived(updates);
    }

    @Override
    public String getBotToken() {

        return "2075246418:AAHnPA1V23H8twB7wwdrKcAJNy77cADJyeE";
    }

    @Override
    public String getBotUsername() {

        return "emojiofy_bot";
    }

    @Override
    public void onRegister() {
        super.onRegister();
    }
}

File: src/main/java/EmojiBot.kt
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */

import org.json.JSONException
import org.json.JSONObject
import org.telegram.telegrambots.bots.TelegramLongPollingBot
import org.telegram.telegrambots.meta.api.methods.send.SendMessage
import org.telegram.telegrambots.meta.api.objects.Update
import org.telegram.telegrambots.meta.exceptions.TelegramApiException
import java.io.*
import java.util.*

/**
 *
 * @author Nakul
 */
class EmojiBot : TelegramLongPollingBot() {

    override fun onUpdateReceived(update: Update) {
        val message = SendMessage()
        message.chatId = update.message.chatId.toString()
        if (update.hasMessage() && update.message.hasText()) {
            val appendText = StringBuilder()

            val array = update.message.text.toString()
                .replace("\"/[^0-9a-zA-Z]/g\"","")
                .split(" ")

            println(array)
           // val holdDetails = HoldDetails()
            val b = StringBuilder()
            //val addEmojisToList = ArrayList<String>()

            for (value in array) {
               // val t = text.split(" ")
                val file = File("C:\\Users\\Nakul\\Documents\\NetBeansProjects\\EmojiCopyPastaBot\\EmojiData.json")
                val reader = BufferedReader(FileReader(file))
                var st: String?

                while (reader.readLine().also {
                        st = it } != null) {
                    b.append(st)
                    // holdDetails.s = b.toString()
                }

                reader.close()

                /*//val values = arrayListOf("this","a")
                val ob = JSONObject(b.toString().trim())
                val ob1: JSONObject = ob.getJSONObject(value.lowercase())

                addEmojisToList.add(ob1.toString())*/
            }
            //val values = arrayListOf("this","a")

            val arr = arrayListOf("")
            array.forEach {
              //  println(it)
                //val isTooCommon = array.any { it in values }
                val emojiPastaText = setModifiedText(it,b)
                appendText.append(emojiPastaText)
            }

                message.text = appendText.toString()
           // message.setText(builder1.toString())
            try {
                execute(message) // Call method to send the message
            } catch (ignored: TelegramApiException) {
            }
        }
    }

    private fun setModifiedText (data:String,b:StringBuilder) :StringBuilder {
        val start = System.currentTimeMillis()
        val appendText = StringBuilder()
        val ob = JSONObject(b.toString())
        try {
            val ob1: JSONObject = ob.getJSONObject(data.lowercase())
            // Apply operation from left to right on each element.

            val hold = data + " " + "${ob1.names().get(getRandomPos(ob1.length()))}" + " "
            return appendText.append(hold)

            /*println(returnString(ob1,data.lowercase()))
            if (ob1.names().length() <= 1 || ob1.names().length() > 1) {
                val hold = data + " " + "${ob1.names().get(getRandomPos(ob1.length()))}" + " "
                return appendText.append(hold)
            }*/

        } catch (e: JSONException) {
        }
        return StringBuilder("$data ")
    }
    
    fun returnString (jsonObject: JSONObject,acc:String): String  {
        if (jsonObject.has(acc)) {
            return acc
        }
    return ""
    }

    override fun onUpdatesReceived(updates: List<Update?>?) = super.onUpdatesReceived(updates)

    override fun getBotToken() = "2075246418:AAHnPA1V23H8twB7wwdrKcAJNy77cADJyeE"

    override fun getBotUsername() = "emojiofy_bot"

    //override fun onRegister() = super.onRegister()
}